crates/net/downloaders/src/headers/metrics.rs

```diff
@@ -0,0 +1,40 @@
+use metrics::Counter;
+use reth_interfaces::p2p::error::DownloadError;
+use reth_metrics_derive::Metrics;
+
+/// The header downloader metrics.
+#[derive(Metrics)]
+#[metrics(scope = "downloaders_headers")]
+pub struct HeaderDownloaderMetrics {
+    /// The number of headers that were successfully returned by the downloader.
+    total_returned: Counter,
+    /// Number of headers that were successfully downloaded
+    total_downloaded: Counter,
+    /// Number of timeout errors while requesting headers
+    timeout_errors: Counter,
+    /// Number of validation errors while requesting headers
+    validation_errors: Counter,
+    /// Number of unexpected errors while requesting headers
+    unexpected_errors: Counter,
```

Comment:
IIRC we got support for labels in `metrics-derive`, right? In that case I think it is nicer to just have an `errors` metric with different labels, e.g. `errors{type="timeout"}`/`errors{type="invalid_header"}` etc.

Reply:
let's do it in a follow up. metrics label UX is a little bit scarce, we need to discuss & improve it